packages/good-design/src/apps/ubi/context/ClaimContext.tsx

```diff
@@ -0,0 +1,233 @@
+    void explorerPollLock.acquire("resetLock");
+
+    setTimeout(async () => {
+      void explorerPollLock.release("resetLock");
+      setRefreshRate(4);
+    }, 300);
+  };
+
+  const onClaim = useCallback(async () => {
+    // todo: handle onboard/upgrade flow?
+
+    // initializes a cycle of claim transactions for all contracts
+    setContract(poolContracts[0]);
+  }, [poolsDetails, onSendTx, poolContracts]);
+
+  const onClaimSuccess = useCallback(async () => {
+    transactionState?.resetState();
+    resetState();
+
+    // should handle what happens after all claims are done (eg. showing a next-task modal)
+    // if nothing is done, it will just silently finish
+    await onSuccess?.();
+  }, [transactionState, onSuccess]);
+
+  const switchChain = useCallback(() => {
+    // 4902: Network is not added, and should be done manually
+    // explanation to user is shown through network modal
+    switchNetwork(SupportedChains[claimedAlt.altChain as keyof typeof SupportedChains]).catch((e: any) => {
+      if (e.code === 4902) {
+        // toggleNetworkModal()
+        //todo: discuss how to handle this
+      }
+    });
+  }, [switchNetwork, claimedAlt]);
+
+  useEffect(() => {
+    resetState();
+  }, [/*used*/ account, /* used */ chainId]);
+
+  //Handle navigation to pre-claim screen
+  useEffect(() => {
+    if (!isEmpty(postClaimPools) || explorerPollLock.isAcquired("resetLock")) return;
+
+    if (!isEmpty(poolContracts) && isEmpty(poolsDetails)) return;
+    const unclaimedPools = getUnclaimedPools(poolsDetails);
+
+    if (
+      account &&
+      isEmpty(preClaimPools) &&
+      !loading &&
+      !isUndefined(claimDetails.hasClaimed) &&
+      (claimDetails.hasClaimed === false || !isEmpty(unclaimedPools))
+    ) {
+      const details: any[] = !claimDetails.hasClaimed ? [{ GoodDollar: [claimDetails] }] : [];
+
+      if (!isEmpty(unclaimedPools)) {
+        details.push(...poolsDetails);
+      }
+
+      setClaimPools(details);
+    }
+  }, [claimDetails, preClaimPools, poolContracts, poolsDetails]);
+
+  //Handle navigation to post-claim screen
+  useEffect(() => {
+    void (async () => {
+      if (explorerPollLock.isAcquired("pollLock") || explorerPollLock.isAcquired("resetLock")) return;
+
+      if (!isEmpty(poolContracts) && isEmpty(poolsDetails)) return;
+
+      const unclaimedPools = getUnclaimedPools(poolsDetails);
+      const noPostClaimPools = isEmpty(postClaimPools);
+      const noUnclaimedPools = isEmpty(unclaimedPools);
+
+      const hasClaimed = claimDetails.hasClaimed === true;
+
+      if (account && hasClaimed && noPostClaimPools && noUnclaimedPools && !loading) {
+        await explorerPollLock.acquire("pollLock");
+
+        const claimTransactionList = await getRecentClaims(
+          account,
+          endpoints,
+          provider ?? library,
+          isArray(poolsDetails) && poolsDetails?.length > 0
+        ).then(res => {
+          explorerPollLock.release("pollLock");
```

Comment:
should be in finally

Reply:
done here: https://github.com/GoodDollar/GoodWeb3-Mono/pull/162/commits/79657860d0f7efa5a7149908f9b0b64b85ef6770